WordPress/Classes/Networking/LoginClient.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+import WordPressAPI
+import AutomatticTracks
+
+actor LoginClient {
+
+    struct ApiDetails {
+        let rootUrl: ParsedUrl
+        let loginUrl: String?
+    }
+
+    private let session: URLSession
+
+    init(session: URLSession) {
+        self.session = session
+    }
+
+    func performLoginAutodiscovery(for url: URL) async throws -> ApiDetails {
+        try await performLoginAutodiscovery(for: url.absoluteString)
+    }
+
+    func performLoginAutodiscovery(for string: String) async throws -> ApiDetails {
+        let internalClient = WordPressLoginClient(urlSession: self.session)
+
+        do {
+            let result = try await internalClient.discoverLoginUrl(for: string)
```

Comment:
Does the service perform enough validations to call it a success? For example, `result.apiDetails.findApplicationPasswordsAuthenticationUrl()` returns an optional. Shouldn't the app log if it's unavailable?

Reply:
This is a good point – I adjusted it in 98d68bc24191c344563cb8ca827f3d924220d282